TripKitUI/core/TransportKitUI/SGTripSegmentsView.m

```diff
@@ -104,17 +104,11 @@ - (void)configureForSegments:(NSArray<id<STKTripSegmentDisplayable>> *)segments
 		}
 		
 		// for the coloured strip
-//    UIColor *color = (allowSubtitles && [segment respondsToSelector:@selector(tripSegmentModeColor)])
-//      ? [segment tripSegmentModeColor]
-//      : nil;
-    
-    UIColor *color = (self.colorCodingTransit && [segment respondsToSelector:@selector(tripSegmentModeColor)])
-    ? [segment tripSegmentModeColor]
-    : nil;
+    UIColor *color = [segment respondsToSelector:@selector(tripSegmentModeColor)] ? [segment tripSegmentModeColor] : nil;
 		
 		// the mode image
     UIImageView * modeImageView = [[UIImageView alloc] initWithImage:image];
-    modeImageView.tintColor = color ?: [SGStyleManager darkTextColor];
+    modeImageView.tintColor = (color != nil && self.colorCodingTransitIcon) ? color : [SGStyleManager darkTextColor];
```

Comment:
Does this now cover all the cases? Only coloring the mode icons if we have a color and the app asks to color code mode icons. To show the color dot next to the mode icon, the check is further down: when we have a color AND we allow subtitle.